app/population/middleware/set-breadcrumb.js

```diff
@@ -0,0 +1,40 @@
+const {
+  isCurrentWeek,
+  isPreviousWeek,
+  isNextWeek,
+} = require('../../../common/helpers/date')
+const {
+  formatDateWithRelativeDay,
+} = require('../../../config/nunjucks/filters')
+
+function setBreadcrumb(req, res, next) {
+  const { date, locationName } = req
+
+  let dateSuffix = ''
+
+  if (isCurrentWeek(date)) {
```

Comment:
For help with navigation, I want the breadcrumb to show Last / This / Next week - but we don't have anything that does this via the i18n files - the currently implementation is inside the `nunjucks/filters` - and hardcodes the string value in place. Also, I'd like to use some brackets.

Any suggestions on a better place for this part of the code to live are welcome.

Reply:
It looks like that filter [uses the locales]() for the current week?

I think the implementation for week navigation was to only specifically use `This week` and not next/previous.

Maybe this could use [`formatDateRangeWithRelativeWeek`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/955759046b2ed6ea625949833d74d6e8497a9254/config/nunjucks/filters.js#L228)?

Reply:
re: filters & locales - my mistake, I was getting `current_week` confused with `Today` - https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/config/nunjucks/filters.js#L185-L195

So yeah, the existing week navigation is only to specifically use `This week` and not `Next/Previous`. I discovered through my working with the dashboard - and after discussion with Andy - that there was a use for seeing this information as it keeps the user grounded while navigating around the populations.

That being said, that's not a good reason to hold this up, so I'll separate that part out, and see if I can get back to it later.

Reply:
Also, `formatDateRangeWithRelativeWeek` does not work on single dates very well. And giving it two dates the same, results in `30 to 30 Nov 2020` - which is also what's not wanted.

`formatDateWithRelativeDay` adds in the year, which is also not what's wanted.

And  as an aside, our version of `_isCurrentWeek` take a single date and turns it into an array of dates for comparison, which seems odd.

I can't help but feel that the filters / date utils needs an overhaul, but I definitely don't have time for that.

I'm going to introduce some local date formatting code in this breadcrumb for now, and see about coming back and cleaning this up later.